src/mongocxx/collection.cpp

```diff
@@ -140,6 +140,17 @@ collection::collection(const database& database, void* collection)
                                     database._impl->client_impl)) {
 }
 
+collection::collection(const collection& c) noexcept : _impl{} {
+    impl tmp {*(c._impl)};
+    _impl = stdx::make_unique<impl>(tmp);
```

Comment:
no need for the temp impl here - as long as you don't `std::move` c's impl, make_unique will copy, not move from it.. Also, you can do all the work in the initializer list i.e.

`collection::collection(const collection& c) noexcept : _impl{stdx::make_unique<impl>(*c._impl) {}`